core/templates/dev/head/domain/exploration/RuleObjectFactory.ts

```diff
@@ -30,9 +30,16 @@ export class Rule {
     this.type = type;
     this.inputs = inputs;
   }
+
   // TODO(#7176): Replace 'any' with the exact type. This has been kept as
   // 'any' because the return type is a dict with underscore_cased keys which
   // gives tslint errors against underscore_casing in favor of camelCasing.
+  getInput(): any {
+    return this.inputs;
+  }
+  getType(): string {
```

Comment:
Maybe move this function to above the comment explaining 'any'? Since the type for this is specified...

Reply:
@aks681 
here I used the type any because in many places it is used as string[] or string[][] or string or object having property.  Should I change the type and change every file?
export interface RuleInputInterface{
  [propName: string ]: any;
}
export interface backEndDict{
  rule_type: string,
  inputs: RuleInputInterface
}